test/bench/etc/generate_documents.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * This script regenerates all benchmark test files
+ */
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { randomBytes, randomInt } from 'node:crypto';
+
+import * as bson from '../../../.';
+import { writeFileSync } from 'fs';
+import { resolve } from 'path';
+
+type numericConstructor =
+  | typeof bson.Double
+  | typeof bson.Long
+  | typeof bson.Int32
+  | typeof bson.Decimal128;
+type numericType = bson.Double | bson.Long | bson.Int32 | bson.Decimal128;
+
+const DOCUMENT_PATH = resolve(`${__dirname}/../documents`);
+
+function generateSimpleTests(typeName: string, newFn: (...arg: any) => any) {
+  const singleField = { a: newFn(0) };
+  const singleElementArray = { a: [newFn(0)] };
+
+  writeFileSync(
+    `${DOCUMENT_PATH}/${typeName}_singleFieldDocument.json`,
+    bson.EJSON.stringify(singleField, { relaxed: false }, 2)
+  );
+  writeFileSync(
+    `${DOCUMENT_PATH}/${typeName}_singleElementArray.json`,
+    bson.EJSON.stringify(singleElementArray, { relaxed: false }, 2)
+  );
+
+  for (const length of [10, 100, 1000]) {
+    let counter = 0;
+    const homogeneousArray = {
+      a: Array.from({ length }, () => newFn(counter++))
+    };
+    writeFileSync(
+      `${DOCUMENT_PATH}/${typeName}_array_${length}.json`,
+      bson.EJSON.stringify(homogeneousArray, { relaxed: false }, 2)
+    );
+  }
+}
+
+function fromNumber(type: numericConstructor, val: number): numericType {
+  switch (type) {
+    case bson.Double:
+      return new type(val);
+    case bson.Long:
+      return type.fromInt(val);
+    case bson.Int32:
+      return new type(val);
+    case bson.Decimal128:
+      return type.fromString(String(val));
+    default:
+      throw new Error('unexpected type');
+  }
+}
+// create numeric type tests
+for (const type of [bson.Int32, bson.Decimal128, bson.Long, bson.Double]) {
+  const typeName = fromNumber(type, 0)._bsontype.toLowerCase();
+  generateSimpleTests(typeName, fromNumber.bind(undefined, type));
+}
+
+// no-argument types
+for (const type of [bson.ObjectId, bson.MaxKey, bson.MinKey]) {
+  const typeName = new type()._bsontype.toLowerCase();
+  generateSimpleTests(typeName, () => new type());
+}
+
+// boolean
+generateSimpleTests('boolean', () => true);
+
+// null
+generateSimpleTests('null', () => null);
+
+// string
+generateSimpleTests('string', () => randomBytes(30).toString('base64'));
+
+// binary
+generateSimpleTests('binary', () => new bson.Binary(randomBytes(30)));
+for (const { name, size } of [
+  { name: 'small', size: 1024 },
+  { name: 'medium', size: 1024 ** 2 }
+]) {
```

Comment:
Note the absence of a large binary document; EJSON stringify fails to stringify documents with Binary instances of size 1024 ** 3 bytes

Reply:
Is this something we should fix?

Reply:
The issue here is that when converting `Binary` instances to EJSON, we convert them to a base64 string. We end up running up against Node's string character limit, so I don't think this is something we can fix easily here.

```
Error: Cannot create a string longer than 0x1fffffe8 characters
    at Object.slice (node:buffer:695:37)
    at Buffer.toString (node:buffer:865:14)
    at Object.toBase64 (/home/wajames/js-bson/src/utils/node_byte_utils.ts:103:54)
    at Binary.toExtendedJSON (/home/wajames/js-bson/src/binary.ts:201:36)
    at serializeDocument (/home/wajames/js-bson/src/extended_json.ts:392:19)
    at serializeValue (/home/wajames/js-bson/src/extended_json.ts:298:58)
    at serializeDocument (/home/wajames/js-bson/src/extended_json.ts:341:23)
    at serializeValue (/home/wajames/js-bson/src/extended_json.ts:298:58)
    at Object.stringify (/home/wajames/js-bson/src/extended_json.ts:475:15)
    at Object.<anonymous> (/home/wajames/js-bson/test/bench/etc/generate_documents.ts:92:16) {
  code: 'ERR_STRING_TOO_LONG'
```